src-ts/lib/tooltip/Tooltip.tsx

```diff
@@ -1,57 +1,73 @@
-import { Dispatch, FC, SetStateAction, useState } from 'react'
+import classNames from 'classnames'
+import { Dispatch, FC, MutableRefObject, ReactNode, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from 'react'
 
-import { ComponentVisible, useHideClickOutside } from '../functions'
-import { IconOutline } from '../svgs'
+import { useClickOutside, UseHoverElementValue, useOnHoverElement } from '../functions'
 
 import styles from './Tooltip.module.scss'
 
 interface TooltipProps {
-    tooltip?: string
+    content?: string
+    positionX?: 'start'|'middle'|'end'
+    positionY?: 'start'|'middle'|'end'
+    trigger: ReactNode
+    triggerOn?: 'click'|'hover'
 }
 
-const Tooltip: FC<TooltipProps> = (props: TooltipProps) => {
+interface ClickHandlersValue {
+    onClick: (ev: any) => void
+}
 
-    const [tooltipOpen, setTooltipOpen]: [boolean, Dispatch<SetStateAction<boolean>>] = useState<boolean>(false)
+function useClickHandlers(trigger: MutableRefObject<any>, toggle: (ev: any) => void): ClickHandlersValue {
+    useClickOutside(trigger.current, () => toggle(false))
+
+    return {
+        onClick: toggle,
+    }
+}
 
+const Tooltip: FC<TooltipProps> = ({
+    content,
+    trigger,
+    triggerOn = 'click',
+    positionX = 'middle',
+    positionY = 'end',
+}: TooltipProps) => {
     // if we didn't get a tooltip, just return an empty fragment
-    if (!props.tooltip) {
+    if (!content) {
         return <></>
     }
 
-    const {
-        isComponentVisible,
-        ref,
-        setIsComponentVisible,
-    }: ComponentVisible = useHideClickOutside(false)
+    const triggerRef: MutableRefObject<any> = useRef(undefined)
+    const [tooltipOpen, setTooltipOpen]: [boolean, Dispatch<SetStateAction<boolean>>] = useState<boolean>(false)
 
-    function toggleOpen(): void {
-        const toggleTo: boolean = !tooltipOpen
-        setTooltipOpen(toggleTo)
-        setIsComponentVisible(toggleTo)
-    }
+    const toggleOpen: (t?: boolean) => void = useCallback((t?: boolean) => {
```

Comment:
Can we do better than one-letter variable names here? I'm used to seeing `<T>` for a generic type, but that's definitely an exception to the rule. What can we use that's better than `t` and `d` here?

Reply:
good call. maybe something like `toggledOpen`?

Reply:
Sure, that works -- and here I was thinking `t` was some sort of type placeholder. Even better that we fix it now that I know. Feel free to fix `d` while you're there!

Reply:
Hey @testflyjets @brooketopcoder ! I updated the variable names, hope these are clear now. Thank you!